LabExT/Movement/PathPlanning.py

```diff
@@ -126,3 +129,158 @@ def _create_outline(
+            self.grid_size)
+        self.cx, self.cy = np.meshgrid(self.x_coords, self.y_coords)
+
+        self.attractive_potenial_field = self.ATTRACTIVE_GAIN * \
+            np.hypot(self.cx - self.target_position.x, self.cy - self.target_position.y)
+        self.potential_field = np.zeros_like(
+            self.cx) + self.attractive_potenial_field
+
+        with self.calibration.perform_in_chip_coordinates():
+            self.start_coordinate = self.calibration.get_position()
+            self.target_position.z = self.start_coordinate.z
+
+            self.current_idx = np.array(
+                [np.argmin(np.abs(self.x_coords - self.start_coordinate.x)),
+                 np.argmin(np.abs(self.y_coords - self.start_coordinate.y))])
+
+            self.current_position = ChipCoordinate(
+                x=self.x_coords[self.current_idx[0]],
+                y=self.y_coords[self.current_idx[1]],
+                z=self.start_coordinate.z)
+
+    def next_waypoint(self) -> list:
+        """
+        Generates iterativly the next waypoint in the given potential field.
+
+        It calculates an infinite number of waypoints with the last waypoints being the actual destination.
+
+        Returns
+        -------
+        Returns a 2D coordinate as next potential field waypoint.
+        """
+        distance_to_target = np.hypot(
+            self.current_position.x - self.target_position.x,
+            self.current_position.y - self.target_position.y)
+
+        if distance_to_target <= self.grid_size:
+            self.current_position = self.target_position
+        else:
+            self.current_idx = self._find_lowest_potential()
+
+            self.current_position = ChipCoordinate(
+                x=self.x_coords[self.current_idx[0]],
+                y=self.y_coords[self.current_idx[1]],
+                z=self.start_coordinate.z)
+
+        return self.current_position
+
+    def set_stage_obstacles(self, *
+                            calibrations: List[Type["Calibration"]]) -> None:
+        """
+        Creates an obstacle in the potential field for each passed calibration.
+
+        Parameters
+        ----------
+        calibrations : List[Calibration]
+            List of calibration instances
+        """
+        self.potential_field = np.zeros_like(
+            self.cx) + self.attractive_potenial_field
+
+        for calibration in calibrations:
+            with calibration.perform_in_chip_coordinates():
+                stage_mask = calibration.stage_polygon.stage_in_meshgrid(
+                    calibration.get_position(),
+                    self.cx,
+                    self.cy)
+
+            for ox, oy in zip(self.cx[stage_mask], self.cy[stage_mask]):
+                o_dist = np.hypot(self.cx - ox, self.cy - oy)
+                field_mask = o_dist < 5 * self.FIBER_RADIUS
+                rep_field = self.REPULSIVE_GAIN * \
+                    (1.0 / o_dist - 1.0 / self.FIBER_RADIUS) ** 2
```

Comment:
I get smoother results in my simulation if the second term in the parentheses is omitted (i.e. assume FIBER_RADIUS -> infty for this line only)

Reply:
I removed the term.